main.tf

```diff
@@ -79,7 +79,7 @@ locals {
   blob_container  = local.create_blob_container ? module.storage.0.container.name : var.blob_container
   storage_account = local.create_blob_container ? module.storage.0.account.name : var.storage_account
   storage_key     = local.create_blob_container ? module.storage.0.account.primary_access_key : var.storage_key
-  queue           = var.use_internal_queue ? "internal://" : "az://${module.storage.0.account.name}/${module.storage.0.queue.name}"
+  queue           = var.use_internal_queue || !local.create_blob_container ? "internal://" : "az://${module.storage.0.account.name}/${module.storage.0.queue.name}"
```

Comment:
```suggestion
  queue           = (var.use_internal_queue || !local.create_blob_container) ? "internal://" : "az://${module.storage.0.account.name}/${module.storage.0.queue.name}"
```

Reply:
I've tested both codes and they work just fine.
I actually didn't get what's the concern with the operation order, there's a single logical expression to be evaluated so what's the point to enclose it in parenthesis?